eng/ApiListing.targets

```diff
@@ -0,0 +1,57 @@
+<Project>
+
+  <Target Name="GenerateApiListing" Condition="'$(IsMainClientLibrary)' == 'true'">
+    <PropertyGroup>
+      <_RefSourceOutputPath>$([System.IO.Directory]::GetParent('$(MSBuildProjectDirectory)'))/api/</_RefSourceOutputPath>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_AllTargetFrameworks Remove="@(_AllTargetFrameworks)" />
+      <_AllTargetFrameworks Include="$(TargetFrameworks);$(TargetFramework)" />
+      <_DeduplicatedTargetFramework Include="%(_AllTargetFrameworks.Identity)" />
+    </ItemGroup>
+    
+    <RemoveDir Directories="$(_RefSourceOutputPath)" />
+
+    <Error Condition="'$(GenerateAPIListing)' != 'true' AND !$(OriginalVersion.Contains('preview'))" Text="GenerateAPIListing must be set to true for non release version" />
+
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
```

Comment:
If you do this with a property instead of a custom target you don't need to multiplex over the target frameworks yourself so this target will be much simplier.

Reply:
Yeah, but I do other things not just multiplexing.

Reply:
What other things are you doing? I'd prefer us not try to mimic this multiplexing as it is complicated to keep working.

Reply:
For example removing the `api`  directory if GenerateAPIListing is disabled to make sure the setting and the listing is in sync.

Reply:
That can be done as another target. You can easily add that as a before target for the shared target.

Reply:
Done